lib/sequent/core/event_publisher.rb

```diff
@@ -0,0 +1,57 @@
+module Sequent
+  module Core
+    #
+    # EventPublisher ensures that, for every thread, events will be published in the order in which they are meant to be published.
```

Comment:
I think "... will be published breadth (or width) first" is better.

To clarify, with breadth first I mean: first all first level events than the consequences of those events.